core/handlers/v2/shutdown_handler.go

```diff
@@ -0,0 +1,32 @@
+package v2
+
+import (
+	"net/http"
+	"os"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/SpectoLabs/hoverfly/core/handlers"
+	"github.com/codegangsta/negroni"
+	"github.com/go-zoo/bone"
+)
+
+type ShutdownHandler struct {
+}
+
+func (this *ShutdownHandler) RegisterRoutes(mux *bone.Mux, am *handlers.AuthHandler) {
+	mux.Get("/api/v2/shutdown", negroni.New(
+		negroni.HandlerFunc(am.RequireTokenAuthentication),
+		negroni.HandlerFunc(this.Get),
+	))
+}
+
+func (this *ShutdownHandler) Get(w http.ResponseWriter, req *http.Request, next http.HandlerFunc) {
+	handlers.WriteResponse(w, []byte(""))
+	go func() {
+		log.Warning("Hoverfly will shut down in 10 seconds")
+		time.Sleep(time.Second * 10)
```

Comment:
Why the sleep? Doesn't `hoverctl stop` need to call this, in which case it should be instant?

This shouldn't be a get request it should be a `DELETE` I think